src/pybind/mgr/dashboard/__init__.py

```diff
@@ -48,7 +48,38 @@ def __getattr__(self, item):
     # raise an ImportError
     import sys
     import mock
-    sys.modules['ceph_module'] = mock.Mock()
 
     mgr = mock.Mock()
     mgr.get_frontend_path.side_effect = lambda: os.path.abspath("./frontend/dist")
+
+    def mock_ceph_modules():
+        class MockRadosError(Exception):
+            def __init__(self, message, errno=None):
+                super(MockRadosError, self).__init__(message)
+                self.errno = errno
+
+            def __str__(self):
+                msg = super(MockRadosError, self).__str__()
+                if self.errno is None:
+                    return msg
+                return '[errno {0}] {1}'.format(self.errno, msg)
+
+        rbd = mock.Mock()
```

Comment:
to minimize the effort to maintain a mock of, for instance python-rbd, probably we could reuse https://github.com/ceph/ceph/blob/master/src/pybind/rbd/mock_rbd.pxi ? currently it's built only if `BUILD_DOC` cython constant is set. see

https://github.com/ceph/ceph/blob/8893f292e9eb45875b4eb7ede1a16f964107de88/src/pybind/rbd/rbd.pyx#L36-L40

Reply:
@tchaikov provided this files does not exist in nautilus/octopus, is it OK for you if we investiate this approach for master/pacific and keep the current behavior as  it is here?